servers/cu/src/domain/worker/evaluate.js

```diff
@@ -158,12 +158,20 @@ export function evaluateWith ({
    *
    * Finally, evaluates the message and returns the result of the evaluation.
    */
-  return ({ streamId, moduleId, wasmModule, moduleOptions, processId, noSave, name, deepHash, cron, ordinate, isAssignment, Memory, message, AoGlobal }) =>
+  return ({ streamId, moduleId, wasmModule, moduleOptions, processId, noSave, name, deepHash, cron, ordinate, isAssignment, Memory, message, AoGlobal, close }) => {
+    /**
+     * If we receive the close flag, it means we have reached the end of the eval stream.
+     * We will remove the WASM instance from the cache and skip loading the module.
+     */
+    if (close) {
+      wasmInstanceCache.delete(streamId)
+      return
+    }
```

Comment:
Could there be a test for this branch in `worker/evaluate.test.js`? Can probably initialize the cache, inject it, call while passing `close` then assert the injected caches size was decremented by 1.